### ISSUE TITLE
Problem formatting .cs files

### ISSUE THREAD

**Issue by gafter** [dotnet/roslyn#1066]
There is a problem while formatting (indenting, adding/erasing whitespaces and/or newlines, etc.) directives such as #endregion, #pragma, etc..

The problem lies within the following classes/functions:
-SyntaxFormatter
-NormalizeWhitespaceCore
-NormalizeWhitespace

These functions are inconsistent with the CSharp compiler, which allows lines such as:

```cs
#endregion foo
```
where the conventional way of doing this is:

```cs
#endregion //foo
```
The formatting erases the whitespace after the directive, giving a non-compilable file like this:

```cs
#endregionfoo <--- here, having written "#endregion //foo" wouldn´t cause any problem, because one would be getting a compilable "#endregion//foo"
```

Similar problem with #pragma, where an extra enter is erased while formatting the following:

```cs
#pragma warning disable 123

namespace foo {
}

#pragma warning restore 123
```

one gets the following non-compilable code:

```cs
#pragma warning disable 123
namespace foo {
}#pragma warning restore 123  <--- extra enter erased here, after the closing bracket
```

**Comment by gafter** [comment]
This bug was originally reported at http://roslyn.codeplex.com/workitem/490

**Comment by Pilchie** [comment]
Per the investigation I did and commented on in the internal bug here (1110300), I was able to repro only using NormalizeWhitespace, which is a core compiler API.

**Comment by gafter** [comment]
NormalizeWhitespace is a thin wrapper over SyntaxFormatter. See https://github.com/dotnet/roslyn/blob/ae1205fe68287242aa71b032deee6965cdfa61f8/src/Compilers/CSharp/Portable/Syntax/CSharpSyntaxNode.cs#L961-L964

**Comment by mattwar** [comment]
Fixed by #2121
